### ISSUE TITLE
[monitoring v2] can't install monitoring on hosted eks/gks/aks cluster

### ISSUE THREAD

**Issue by slickwarren** [rancher/rancher#32201]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
* deploy eks cluster with enough resources for monitoring
* deploy monitoring 

**Result:**
`rancher-monitoring-operator` is in crashloopback error with 
```
level=info ts=2021-04-16T23:22:03.20894875Z caller=main.go:233 msg="Starting Prometheus Operator" version="(version=0.46.0, branch=refs/tags/pkg/client/v0.46.0, revision=7f94a06b86d41c20176f0d5b53aa0100fdc361e6)"
level=info ts=2021-04-16T23:22:03.209004256Z caller=main.go:234 build_context="(go=go1.15.8, user=s-urbaniak, date=20210224-17:30:02)"
listening failed:10250listen tcp :10250: bind: address already in use

```

**Other details that may be helpful:**
monitoring times out waiting on that pod to be healthy again

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.5.8-rc4
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Hosted GKE and EKS
- Machine type (cloud/VM/metal) and specifications (CPU/memory): 10vCPU 
- Kubernetes version (use `kubectl version`):  v1.17.17-gke.3000     v1.18.9-eks-d1db3c

- Docker version (use `docker version`): gke:  19.3.6 eks: 19.3.13

**Comment by thehejik** [comment]
I can confirm this issue on GKEv2 with `rancher-monitoring-14.5.100-rc07` v2 chart and with rancher:v2.5-head `41f173a`

**Comment by slickwarren** [comment]
this also happens for AKS

**Comment by aiyengar2** [comment]
@slickwarren @thehejik can you set prometheusOperator.hostNetwork=false and try again? We have a UI ticket out for this that is currently being worked on https://github.com/rancher/dashboard/issues/2573

**Comment by slickwarren** [comment]
That fixed the issue on my aks cluster, trying now on gke/eks

**Comment by slickwarren** [comment]
[this does work ](https://github.com/rancher/dashboard/issues/2573)for aks and eks and GKE on a clean install of monitoring. however, when upgrading a GKE cluster with the new settings, I ran into the following issue: 
```
Error: UPGRADE FAILED: failed to create resource: Internal error occurred: failed calling webhook "prometheusrulemutate.monitoring.coreos.com": Post https://rancher-monitoring-operator.cattle-monitoring-system.svc:443/admission-prometheusrules/validate?timeout=10s: no endpoints available for service "rancher-monitoring-operator"
```

**Comment by aiyengar2** [comment]
Also, for users who are encountering this issue, as a TLDR here are some workarounds you can consider:
1. For AKS / EKS Clusters, set `prometheusOperator.hostNetwork=false`. 
- As noted in https://github.com/rancher/rancher/issues/31786 and https://github.com/rancher/dashboard/issues/2573, setting `prometheusOperator.hostNetwork=true` for managed Kubernetes clusters may not be necessary unless you are specifically using a custom CNI.
2. For GKE clusters or other managed Kubernetes clusters that are using a custom CNI, the latest version of the Monitoring chart release in 2.5.8 (e.g. version 14.5.100) will support the new field [`prometheusOperator.tls.internalPort` from the upstream kube-prometheus-stack chart](https://github.com/prometheus-community/helm-charts/blob/5cfd0de9b6a145299f93b93ce58d3ba5a3fb4212/charts/kube-prometheus-stack/values.yaml#L1322-L1323). This can be overridden to be set to a different port (e.g. 10251) to avoid port conflicts but will require you to open additional firewall rules to enable access to it
- Note: This TLS port needs to be opened because the Kubernetes API will hit that endpoint on creating PrometheusRule resources in order to satisfy the Admissions Webhooks deployed by the chart.
3. Alternatively, you can bypass the error entirely by just disabling admission webhooks, e.g. set [`prometheusOperator.admissionsWebhooks=false`](https://github.com/prometheus-community/helm-charts/blob/5cfd0de9b6a145299f93b93ce58d3ba5a3fb4212/charts/kube-prometheus-stack/values.yaml#L1329)

**Comment by thehejik** [comment]
@aiyengar2 Please is following expected? 

When deploying latest v2 `rancher-monitoring-14.5.100` on GKE with default values (clusterType `Google GKE` selected) I noticed that `prometheusOperator.tls.internalPort` key is still set to port `10250` in value.yaml. Anyway `rancher-monitoring-operator` pod is listening on the new port 8443 after (successful) deployment. Is this expected or the value from `internalPort` is somehow ignored or overwritten?

**Comment by slickwarren** [comment]
when using v2.5.8-rc11 on monitoring v14.5.100:
GKE, EKS and AKS -- clean install of 14.5.100 -- pass
GKE -- upgrade from 9.4.203 -> 14.5.100 -- intermittent failures filed https://github.com/rancher/rancher/issues/32416 
EKS -- upgrade from 9.4.203 -> 14.5.100 -- pass 
AKS -- upgrade from 9.4.203 -> 14.5.100 -- pass


closing this issue as there seems to only be a specific failure here

**Comment by aiyengar2** [comment]
@tirelibirefe I looked at the values.yaml that have been released:
- [14.5.100 (last Monitoring V2 released in 2.5.x before 2.6.0 was released)](https://github.com/rancher/charts/blob/7bd8cb6123ec5ce9c4aec7f14917de6b3dd74370/charts/rancher-monitoring/rancher-monitoring/14.5.100/values.yaml#L1769)
- [100.0.0+up16.6.0 (latest Monitoring V2 released in Rancher 2.6.x)](https://github.com/rancher/charts/blob/7bd8cb6123ec5ce9c4aec7f14917de6b3dd74370/charts/rancher-monitoring/rancher-monitoring/100.0.0%2Bup16.6.0/values.yaml#L1943)
- [16.6.0 (latest Monitoring V2 released in Rancher 2.5.x, tracking the changes in 100.0.0+up16.6.0)](https://github.com/rancher/charts/blob/7bd8cb6123ec5ce9c4aec7f14917de6b3dd74370/charts/rancher-monitoring/rancher-monitoring/16.6.0/values.yaml#L1943)

All of these show `prometheusOperator.tls.internalPort` have been set to `8443`, not `10250`, so it doesn't seem like we regressed on the fix for this issue (on a surface level; I could be wrong and this may require deeper investigation 😄).

So before we investigate this, can you double check what value `prometheusOperator.tls.internalPort` is set to in your current deployed chart? If it is still `10250`, can you update it to `8443`? That should fix the issue; perhaps the fix was not absorbed on a `helm upgrade` for some reason.

If not, can you provide more information by filling out the same template filled out in the issue description so we can test it out to see if we can reproduce? I assume you might be on a different Rancher version, Monitoring version, etc. so we need that to faithfully reproduce this issue.

**Comment by vyom-soft** [comment]
Hello,
I installed latest https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack
I am on MetalLB, KubeSpray, Ingress-nginx.

with hostNetwork: true in the value.yaml
prometheusOperator.enabled=true
internalPort=10250
<img width="1134" alt="image" src="https://user-images.githubusercontent.com/18686082/177750365-dd36c954-af71-46e1-9155-4ba67a2c1c49.png">

Any way to overcome this.
